### PR TITLE
Dockerize local development

### PR DESCRIPTION
**Pull request by geriom**
# Changes

Running the site locally requires installing a lot of dependencies
spanning Python, Go, and Node packages; this can be daunting. Dockerizing
the website reduces the dependencies to one, in a reproducible
environment.

I used the same base image as the docy-example site. Two images are
created, the first one installs the python requirements for the sync
script, then runs the sync script. The second one runs the website and
mounts the folder as a volume inside the container.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -16,40 +16,81 @@
 
 ## Running Locally
 
-Step 1
```
Comment (bobcatfish) [inline DEVELOPMENT.md:19]: might make sense to move the "dependencies" section above into the "running natively" section, since you won't need those dependencies to go the docker based route :D

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -16,40 +16,81 @@
...
+   netlify dev
+   ```
+
+1. Verify that the website is working
+
+   Open your web browser and type `http://localhost:8888` in the navigation bar.
+   This opens a local instance of the website, you can now make changes in the
+   documentation and those changes will immediately show up in the browser after
+   you save.
+
```
Comment (bobcatfish) [inline DEVELOPMENT.md:93]: totally some scope creep but if we could just get `netlify dev` to work from the docker container as well we could remove this whole section.

on the other hand, running the hugo server is probably good enough, and maybe with the new docker based instructions we could just have a reference to `netlify dev` as another option and remove the section anyway?

**Review by bobcatfish — COMMENTED** [review]
Wow this experience is SO MUCH BETTER, thanks @geriom !!! 🤩 

/approve

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+FROM klakegg/hugo:ext-alpine as dependencies
+
+WORKDIR /app
+RUN apk add git gcc build-base python3-dev py3-pip
+COPY . /app
+RUN npm install
+RUN python3 -m venv .venv
```
[the remark this answers is not in the document]
Comment (geriom) [reply Dockerfile:7]: I added a `.dockerignore` to ignore `.venv` and `node_modules`.

File: DEVELOPMENT.md (hunk 1 above)
Comment (geriom) [reply DEVELOPMENT.md:19]: Reorganized the sections, with "Prerequisites" and "Setup" subsections for docker and native.

File: DEVELOPMENT.md (hunk 2 above)
Comment (geriom) [reply DEVELOPMENT.md:95]: I can update the Dockerfile in a subsequent PR to make this work.
